src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -4719,13 +4719,19 @@ void ClassFileParser::verify_legal_class_modifiers(jint flags, const char* name,
   const bool is_annotation = (flags & JVM_ACC_ANNOTATION) != 0;
   const bool major_gte_1_5 = _major_version >= JAVA_1_5_VERSION;
 
+  const bool valid_value_class = is_identity || is_interface ||
```

Comment:
Why is `is_identity` included in `valid_value_class`?  I would have expected identity to flag a class as an invalid value class.  Maybe this is just a naming issue with the `valid_value_class` variable?

Reply:
Yes it is a naming issue. It really means "if this is a value class then it is a valid value class."